src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -3079,6 +3080,45 @@ class spell_pri_shadow_mend_periodic_damage : public AuraScript
     }
 };
 
+// 32379 - Shadow Word: Death
+class spell_pri_shadow_word_death : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellEffect
+        ({
+            { SPELL_PRIEST_SHADOW_WORD_DEATH, EFFECT_1 },
+            { SPELL_PRIEST_SHADOW_WORD_DEATH, EFFECT_2 },
+            { SPELL_PRIEST_SHADOW_WORD_DEATH, EFFECT_4 }
+        });
+    }
+
+    void HandleEffectHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+        if (!caster || !target)
```

Comment:
both never null (unless hit entity can be gameobject, which isn't the case here)

Reply:
so no need for the check, right?

Reply:
correct. you can use the variables instead of `GetCaster()` or `GetHitUnit()` in code below, but you dont need the null checks